### PR TITLE
feat: add mqtt-packet defs

### PR DESCRIPTION
**Pull request by jedwards1211**
- Links to documentation: https://github.com/mqttjs/mqtt-packet/blob/master/types/index.d.ts
- Link to GitHub or NPM: https://github.com/mqttjs/mqtt-packet
- Type of contribution: new definition

Other notes:
I did correct a few of the types from TS by inspecting the JS code

### CODE REVIEW

File: definitions/npm/mqtt-packet_v6.6.x/flow_v0.84.x-/mqtt-packet_v6.6.x.js [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      qos?: QoS,
+      retain?: boolean,
+      properties?: {|
+        willDelayInterval?: number,
+        payloadFormatIndicator?: number,
+        messageExpiryInterval?: number,
+        contentType?: string,
+        responseTopic?: string,
+        correlationData?: Buffer,
+        userProperties?: Object,
```
Comment (pascalduez) [inline definitions/npm/mqtt-packet_v6.6.x/flow_v0.84.x-/mqtt-packet_v6.6.x.js:55]: As you probably know `Object` is now aliased to `any`.

Comment (jedwards1211) [reply definitions/npm/mqtt-packet_v6.6.x/flow_v0.84.x-/mqtt-packet_v6.6.x.js]: Whoops, forgot to replace those from the TS defs.  I replaced them with `{ ... }` now

**Comment by jedwards1211** [comment]
Oh hang on, I should also get rid of the remaining dumb `I` prefixes

**Comment by jedwards1211** [comment]
Okay ready

**Review by pascalduez — APPROVED** [review]

**Pull request merged**